syntax = "proto2";

package vm_scheduler.proto;

import "structures/agent.proto";
import "structures/job.proto";
import "structures/task.proto";

import "google/protobuf/empty.proto";


service AgentApiScheduler {
  rpc getAssignedJobs(VmId) returns (AssignedJobs);

  rpc getJobToLaunch(LaunchRequest) returns (JobToLaunch);

  rpc updateJobState(ExecutionJobState) returns (google.protobuf.Empty);
}
